test/clients.go

```diff
@@ -38,8 +38,15 @@ type KubeClient struct {
 }
 
 // NewSpoofingClient returns a spoofing client to make requests
-func NewSpoofingClient(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool, opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
-	return spoof.New(client.Kube, logf, domain, resolvable, Flags.IngressEndpoint, opts...)
+func NewSpoofingClient(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool,
+	opts ...spoof.TransportOption) (*spoof.SpoofingClient, error) {
+	return NewSpoofingClientWithOptions(client, logf, domain, resolvable, spoof.GetDefaultSpoofingClientOptions(), opts...)
+}
+
+// NewSpoofingClientWithOptions returns a spoofing client to make requests with spoofing options
+func NewSpoofingClientWithOptions(client *KubeClient, logf logging.FormatLogger, domain string, resolvable bool,
```

Comment:
It's kind of strange to have `spoof.SpoofingClientOptions, ...spoof.TransportOption`.

What about the following:

```golang
func NewSpoofingClient2(*KubeClient, logging.FormatLogger, string, bool, ...spoof.ClientOptions) (*spoof.SpoofingClient, error) {
  ...
`

And then:

```golang
type ClientOptions func(*SpoofingClient) *SpoofingClient

func HTTPOption(opt TransportOption) ClientOption {
  return func(s *SpoofingClient) *SpoofingClient {
    s.Client.Transport = opt(s.Client.Transport)
    return s
  }
}
```

You can then convert calls to `NewSpoofingClient` to `NewSpoofingClient2` at your leisure. Once all calls are converted, you can redefine `NewSpoofingClient` to the same call signature as the "2" version if desired.